main.py

```
@@ -0,0 +1,243 @@
+
+
+        token_bytes = token.encode('utf-8')  # Convert token to bytes if it's a string
+
+        payload = jwt.decode(token_bytes, SECRET_KEY, algorithms=[ALGORITHM])
+        username = payload.get("sub")
+        if username is None:
+            raise HTTPException(status_code=401, detail="Invalid authentication credentials")
+        return User(username=username)
+    except jwt.PyJWTError as e:
+        logger.error(f"JWT error: {e}")
+        raise HTTPException(status_code=401, detail="Invalid authentication credentials")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)